README.md

```diff
@@ -1 +1,311 @@
+
+  > During the initial authorization, scopes found in the resources configuration will be concatenated and added as the `scope` parameter to the request. This means that even if the `useOpenId` parameter is set to false, but one of the resources has `openid` scope associated with it, OIDC authentication will be requested. In this case, the `openid` scope will also be included in authorization request made for the individual resource, resulting in ID Token being present in the resource-specific token set.
+
+* `usePKCE`
+
+  The Authorization Code grant is the recommended _authorization_ flow for all types of OAuth 2.0 clients interacting directly with the Resource Owner's user-agent. According to the [OAuth 2.0 Security BCP](https://tools.ietf.org/html/draft-ietf-oauth-security-topics-13#section-3.1.1), OAuth 2.0 clients MUST use the [PKCE](https://tools.ietf.org/html/rfc7636) extension to the Authorization Code grant and the authorization servers MUST support it. PKCE is cheap to implement and allows both the Client and the Authorization Server to verify the authorization code before it is exchanged for tokens. The default value for this setting is `true` and it is recommended to keep it that way.
+
+  > If an Authorization Server does not support PKCE, providing extra parameters should not introduce any negative effects, for unrecognized query parameters MUST be ignored at the [authorization](https://tools.ietf.org/html/rfc6749#section-3.1) and the [token](https://tools.ietf.org/html/rfc6749#section-3.2) endpoints.
+
+* `useMasterAccessToken`
+
+  If it fails to obtain a valid access token for a resource, with this option set to `true`, the helper's `fetch` will proceed to the protected resource with the master access token. It is not a recommended way to make requests to a protected API, for it would probably require a long lived almighty access token which will be shared with the resource (server). In addition, this will discard the "audience restricted" paradigm. This setting was added for flexibility to account for possible special situations when no distinct resources can be identified. The default value for this setting is a `false`.
+
+* `sessionKey`
+
+  The helper aims to maintain the authorization state via the back-channel and never expose the tokens to the client side. In order to be able to manage authorization state specific to a Resource Owner, it needs access to the user session presumably being managed by a framework. It is assumed that the session object can be identified by a key and accessed when handling a request. The setting represents this key and its default value is `'session'`.
+
+  If the user session cannot be found, the authorization state will be stored in a local (private) variable and shared across all users of the helper instance. This may be a desirable effect in an API proxy of sorts.
```

Comment:
I'm curious what this means. How it is shared? I'd like to know more about why this may be desirable. Definitely need more elaboration on this mode of operation.